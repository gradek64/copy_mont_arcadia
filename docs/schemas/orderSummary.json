{
  "type": "object",
  "properties": {
    "basket": {
      "$ref": "#/definitions/basket"
    },
    "deliveryLocations": {
      "type": "array",
      "items": {
        "type": "object",
        "properties": {
          "deliveryLocationType": {
            "type": "string",
            "example": "HOME"
          },
          "label": {
            "type": "string",
            "example": "Home Delivery Standard (UK up to 4 working days; worldwide varies) Express (UK next or nominated day; worldwide varies)"
          },
          "selected": {
            "type": "boolean",
            "example": false
          },
          "enabled": {
            "type": "boolean",
            "example": false
          },
          "errorCode": {
            "type": "string",
            "example": "_API_CANT_RESOLVE_FFMCENTER.2"
          },
          "deliveryMethods": {
            "type": "array",
            "items": {
              "type": "object",
              "properties": {
                "shipModeId": {
                  "type:": "number",
                  "example": 26504
                },
                "shipCode": {
                  "type": "string",
                  "example": "S"
                },
                "deliveryType": {
                  "type": "string",
                  "example": "HOME_STANDARD"
                },
                "label": {
                  "type": "string",
                  "example": "Standard Delivery"
                },
                "additionalDescription": {
                  "type": "string",
                  "example": "2 - 4 Working days"
                },
                "cost": {
                  "type": "number",
                  "example": 0
                },
                "selected": {
                  "type": "boolean",
                  "example": true
                },
                "enabled": {
                  "type": "boolean",
                  "example": false
                },
                "deliveryOptions": {
                  "type": "array",
                  "items": {
                    "type": "object",
                    "properties": {
                      "shipModeId": {
                        "type": "number",
                        "example": 28004
                      },
                      "dayText": {
                        "type": "string",
                        "example": "Wed"
                      },
                      "dateText": {
                        "type": "string",
                        "example": "18 Oct"
                      },
                      "nominatedDate": {
                        "type": "string",
                        "example": "2017-10-18"
                      },
                      "price": {
                        "type": "string",
                        "example": "6.00"
                      },
                      "enabled": {
                        "type": "boolean",
                        "example": false
                      },
                      "selected": {
                        "type": "boolean",
                        "example": true
                      }
                    }
                  }
                }
              }
            }
          }
        }
      }
    },
    "giftCards": {
      "$ref": "#/definitions/giftCards"
    },
    "deliveryInstructions": {
      "type": "string",
      "example": "Leave under the porch."
    },
    "smsMobileNumber": {
      "type": "string",
      "example": ""
    },
    "shippingCountry": {
      "type": "string",
      "example": "United Kingdom"
    },
    "savedAddresses": {
      "$ref": "#/definitions/savedAddresses"
    },
    "ageVerificationDeliveryConfirmationRequired": {
      "type": "boolean",
      "example": false
    },
    "estimatedDelivery": {
      "type": "array",
      "items": {
        "type": "string",
        "example": "No later than Monday 23 October 2017"
      }
    },
    "cardNumberHash": {
      "type": "string",
      "example": "tjOBl4zzS+ueTZQWartO5l968iOmCOix"
    },
    "creditCard": {
      "$ref": "#/definitions/creditCard"
    },
    "billingDetails": {
      "$ref": "#/definitions/userDetails"
    },
    "deliveryDetails": {
      "$ref": "#/definitions/userDetails"
    },
    "isGuestOrder": {
      "type": "boolean",
      "example": true
    }
  }
}
