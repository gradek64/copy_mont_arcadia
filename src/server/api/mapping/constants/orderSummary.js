export const orderSummaryConstants = ({ storeId, catalogId }) => ({
  page: '',
  URL:
    'OrderCalculate?updatePrices=1%26orderItemId*=%26quantity*=%26URL=BillingAddressView',
  returnPage: 'ShoppingBag',
  outOrderItemName: '',
  Proceed: '',
  CheckoutURL:
    'OrderCopy?URL=OrderPrepare%3fURL%3dOrderDisplay%26errorViewName=InvalidInputErrorView',
  PromoURL:
    'PromotionCodeManage?URL=OrderCalculate%3fURL%3dOrderPrepare%3fURL%3dOrderItemDisplay%26taskType=A%26errorViewName=OrderItemDisplayViewShiptoAssoc',
  shoppingBasketURL: `OrderCalculate?langId=-1%26storeId=${storeId}%26catalogId=${catalogId}%26updatePrices=1%26calculationUsageId=-1%26URL=OrderItemDisplay`,
  showCheckout: false,
})

export const guestOrderSummaryConstants = {
  new: 'Y',
  returnPage: 'ShoppingBag',
}

export const deliveryMethodsMonty = {
  homeStandard: 'HOME_STANDARD',
  homeExpress: 'HOME_EXPRESS',
  storeExpress: 'STORE_EXPRESS',
  storeStandard: 'STORE_STANDARD',
  storeImmediate: 'STORE_IMMEDIATE',
  parcelshopCollection: 'PARCELSHOP_COLLECTION',
}

export const deliveryMethods = {
  Standard: deliveryMethodsMonty.homeStandard,
  S: deliveryMethodsMonty.homeStandard,
  E: deliveryMethodsMonty.homeExpress,
  'Retail Store Express': deliveryMethodsMonty.storeExpress,
  'Retail Store Standard': deliveryMethodsMonty.storeStandard,
  'Retail Store Immediate': deliveryMethodsMonty.storeImmediate,
  'Retail Store Collection': deliveryMethodsMonty.parcelshopCollection,
}

export const deliveryLocationDescriptions = {
  HOME: ['homeDeliveryText', 'homeDeliveryContent'],
  STORE: ['storeDeliveryText', 'collectFromStoreContent'],
  PARCELSHOP: ['hermesDeliveryText', 'hermesContent'],
}

export const addDeliveryAddressConstants = {
  URL: 'ProcessDeliveryDetails',
  proceed: '',
  registerType: 'R',
  returnPage: 'ShoppingBag',
  isoCode: '',
  page: 'account',
  editRegistration: 'Y',
  editSection: '',
  outOrderItemName: '',
  actionType: 'updateCountryAndOrderItems',
  shipping_errorViewName: 'UserRegistrationForm',
  lookupHouseNumber: '',
  lookupPostcode: '',
  errorViewName: 'AddressUpdateAjaxView',
  preferredLanguage: '',
  preferredCurrency: '',
  sourcePage: '',
  montyUserAction: 'shipping',
  deliveryOptionType: 'H',
  preventAddressOverride: 'Y',
  addressResults: '',
}
