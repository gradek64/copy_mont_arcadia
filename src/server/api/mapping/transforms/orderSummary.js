import basketTransform from './basket'
import { detailsFragment } from './logon'
import { path, isEmpty } from 'ramda'
import {
  deliveryMethods,
  deliveryLocationDescriptions,
  deliveryMethodsMonty,
} from '../constants/orderSummary'
import { extractEncodedHTMLText } from '../../utils'
import espotsDesktopConstants from '../../../../shared/constants/espotsDesktop'
import { error } from '../../../lib/logger'

const checkIfBasketObjectWithError = (basket = {}) => {
  const products = path(['products', 'Product'], basket)
  if (
    (Array.isArray(products) &&
      products.find(
        (product) =>
          (product && product.outOfStock) ||
          (product && product.exceedsAllowedQty)
      )) ||
    basket.messageForBuyer
  ) {
    return basket
  }
}

const getDeliveryType = (deliveryLocations = []) => {
  if (Array.isArray(deliveryLocations)) {
    const location = deliveryLocations.find((obj = {}) => obj.selected === true)
    return location ? location.deliveryLocationType : false
  }
  return false
}

const deliveryPriceFragment = (price = '') => {
  if (price !== 0 && (!price || !Number(price))) return false
  return typeof price === 'string' ? Number(price).toFixed(2) : price.toFixed(2)
}

const expressDeliveryOptionFragment = ({
  shipModeId = '',
  dayText = '',
  dateText = '',
  nominatedDate = '',
  price = 'N/A',
  selected = '',
  enabled = false,
}) => ({
  dayText,
  dateText,
  nominatedDate,
  selected: selected === 'true' || selected === true,
  shipModeId: Number(shipModeId) || 0,
  price: deliveryPriceFragment(price) || 'N/A',
  enabled,
})

const expressDeliveryFragment = ({
  label = '',
  additionalDescription = '',
  deliveryDates = {},
  selected = '',
  enabled = false,
  price,
  shipCode,
}) => ({
  label,
  additionalDescription,
  selected: selected || false,
  deliveryType: deliveryMethodsMonty.homeExpress,
  deliveryOptions: Array.isArray(deliveryDates.deliveryOptions)
    ? deliveryDates.deliveryOptions
        .map(expressDeliveryOptionFragment)
        .filter((option) => !isNaN(option && option.price))
    : [],
  enabled,
  cost: typeof price === 'number' && price.toFixed(2),
  shipCode: typeof shipCode === 'string' && shipCode,
})

const deliveryMethodFragment = ({
  shipModeId = '',
  shipCode = '',
  label = '',
  selected = false,
  additionalDescription = '',
  deliveryOptions = [],
  enabled = false,
  price = null,
  nominatedDate,
  estimatedDeliveryDate = '',
}) => {
  const deliveryMethod = {
    shipModeId: Number(shipModeId) || 0,
    deliveryType: deliveryMethods[shipCode] || '',
    label,
    additionalDescription,
    selected,
    deliveryOptions,
    enabled,
    cost: deliveryPriceFragment(price) || 0,
    estimatedDeliveryDate,
  }

  // Match current scrApi response. ShipCodes are only present for these delivery methods
  if (
    shipCode === 'S' ||
    shipCode === 'Retail Store Standard' ||
    shipCode === 'Retail Store Collection'
  ) {
    deliveryMethod.shipCode = shipCode
  }

  if (typeof enabled !== 'undefined') {
    deliveryMethod.enabled = enabled
  }

  if (nominatedDate) {
    deliveryMethod.additionalDescription = `Collection date ${nominatedDate}`
  } else {
    deliveryMethod.additionalDescription = additionalDescription
  }

  return deliveryMethod
}

/**
 * The following logic is required to investigate ADP-195
 */
const checkForInvalidDeliveryMethodData = ({
  deliveryMethods,
  expressDelivery,
  transformedArray,
}) => {
  const hasDeliveryMethods =
    (Array.isArray(deliveryMethods) && deliveryMethods.length) ||
    expressDelivery

  const hasSelectedMethod = (methods) =>
    methods.find((method) => path(['selected'], method) === true)

  const hasSelectedDeliveryMethods = hasSelectedMethod(
    deliveryMethods.concat(expressDelivery)
  )

  if (hasDeliveryMethods && !hasSelectedDeliveryMethods) {
    error(
      'ADP-195 checkForInvalidDeliveryMethodData() Invalid deliveryMethods data: neither deliveryMethods nor expressDelivery are selected',
      {
        deliveryMethods,
        expressDelivery,
      }
    )
  }

  if (
    hasDeliveryMethods &&
    hasSelectedDeliveryMethods &&
    !hasSelectedMethod(transformedArray)
  ) {
    error(
      'ADP-195 checkForInvalidDeliveryMethodData() Invalid deliveryMethods data: neither deliveryMethods nor expressDelivery are selected after transformation',
      {
        deliveryMethods,
        expressDelivery,
        transformedArray,
      }
    )
  }
}
/**
 * end of ADP-195 logic
 */

const deliveryMethodsFragment = ({ deliveryMethods = [], expressDelivery }) => {
  if (!Array.isArray(deliveryMethods)) {
    error(
      `ADP-195 deliveryMethodsFragment() deliveryMethods is not an array, is actually of type ${typeof deliveryMethods}`,
      deliveryMethods
    )
  }

  if (Array.isArray(deliveryMethods)) {
    let shouldDumpArray = false

    deliveryMethods.forEach((deliveryMethod, index) => {
      if (deliveryMethod === undefined) {
        shouldDumpArray = true
        error(
          `ADP-195 deliveryMethodsFragment() deliveryMethod[${index}] is undefined, expected an object`
        )
      } else if (deliveryMethod === null) {
        shouldDumpArray = true
        error(
          `ADP-195 deliveryMethodsFragment() deliveryMethod[${index}] is null, expected an object`
        )
      } else if (Array.isArray(deliveryMethod)) {
        shouldDumpArray = true
        error(
          `ADP-195 deliveryMethodsFragment() deliveryMethod[${index}] is an array, expected an object`
        )
      } else if (Object.keys(deliveryMethod).length === 0) {
        shouldDumpArray = true
        error(
          `ADP-195 deliveryMethodsFragment() deliveryMethod[${index}] is an object with no keys`
        )
      }
    })

    if (shouldDumpArray) {
      error(
        'ADP-195 deliveryMethodsFragment() deliveryMethods failed array member type validation',
        deliveryMethods
      )
    }
  }

  const deliveryMethodsArray = deliveryMethods.map(deliveryMethodFragment)

  if (expressDelivery) {
    if (Array.isArray(expressDelivery)) {
      error(
        `ADP-195 deliveryMethodsFragment() expressDelivery is an array, expected an object`
      )
    } else if (Object.keys(expressDelivery).length === 0) {
      error(
        `ADP-195 deliveryMethodsFragment() expressDelivery is an object with no keys`
      )
    }

    deliveryMethodsArray.push(expressDeliveryFragment(expressDelivery))
  }

  checkForInvalidDeliveryMethodData({
    deliveryMethods,
    expressDelivery,
    transformedArray: deliveryMethodsArray,
  })

  return deliveryMethodsArray
}

const deliveryLocationLabelFragment = (
  location = '',
  deliveryLocationTypeForm = {}
) => {
  const textKey = path([location, 0], deliveryLocationDescriptions)
  const contentKey = path([location, 1], deliveryLocationDescriptions)
  if (!textKey || !contentKey) return ''
  return (
    `${deliveryLocationTypeForm[textKey]} ${extractEncodedHTMLText(
      deliveryLocationTypeForm[contentKey]
    )}` || ''
  )
}

const deliveryLocationMethodsFragment = (selected, deliveryoptionsform) => {
  if (selected) {
    if (typeof selected !== 'boolean') {
      error(
        `ADP-195 deliveryLocationMethodsFragment() 'selected' is truthy but not true, is actually of type ${typeof selected}`,
        selected
      )
    }

    return deliveryMethodsFragment(deliveryoptionsform)
  }
  return []
}

const deliveryLocationFragment = (
  { deliveryLocationType = '', selected = false, enabled = false },
  deliveryLocationTypeForm,
  deliveryoptionsform
) => ({
  deliveryLocationType,
  selected,
  enabled,
  // Label is simply the deliveryLocationType when there are no other delivery locations available
  label: deliveryLocationTypeForm
    ? deliveryLocationLabelFragment(
        deliveryLocationType,
        deliveryLocationTypeForm
      )
    : deliveryLocationType,
  deliveryMethods: deliveryLocationMethodsFragment(
    selected,
    deliveryoptionsform
  ),
})

const deliveryLocationsFragment = (
  deliveryLocationTypeForm,
  deliveryoptionsform
) => {
  const { deliveryLocations } = deliveryLocationTypeForm
  if (Array.isArray(deliveryLocations)) {
    return deliveryLocations.map((deliveryLocation) =>
      deliveryLocationFragment(
        deliveryLocation,
        deliveryLocationTypeForm,
        deliveryoptionsform
      )
    )
  }
  // The deliveryLocations array won't exist for international stores.
  // When that's the case, a default 'HOME' location must be generated instead.
  // This is done to match current scrAPI behaviour.
  return [
    deliveryLocationFragment(
      { deliveryLocationType: 'HOME', selected: true, enabled: true },
      null,
      deliveryoptionsform
    ),
  ]
}

const deliveryDetailsFragment = (deliveryoptionsform) => {
  const details = path(['deliveryDetails'], deliveryoptionsform)
  const addressDetailsId = Number(
    path(['selectedAddressID'], deliveryoptionsform)
  )
  return details && addressDetailsId
    ? { ...detailsFragment(details), addressDetailsId }
    : details
      ? detailsFragment(details)
      : null
}

const creditCardFragment = ({ type = '', cardNumberStar = '' } = {}) => ({
  type,
  cardNumberStar,
})

const addressNameFragment = (address = []) => {
  if (!Array.isArray(address)) return ''
  const filteredAddress = address.filter(
    (line) => line && typeof line === 'string'
  )
  if (filteredAddress.length) {
    const mappedAddress = filteredAddress
      .map((line) => line)
      .toString()
      .replace(/,/g, ', ') // foo,bar -> foo, bar
    return mappedAddress.length > 37
      ? mappedAddress.slice(0, 37).concat('...')
      : mappedAddress
  }
  return ''
}

const savedAddressFragment = ({
  addressId = 0,
  address1 = '',
  address2 = '',
  city = '',
  state = '',
  postcode = '',
  title = '',
  firstName = '',
  lastName = '',
  country = '',
  telephone = '',
  selected = false,
}) => ({
  id: addressId,
  addressName: addressNameFragment([address2, city, state, postcode, country]),
  selected,
  address1,
  address2,
  city,
  state,
  country,
  telephone,
  postcode,
  title,
  firstName,
  lastName,
})

const estimatedDeliveryFragment = ({ estimatedDelivery = '' } = {}) =>
  estimatedDelivery ? [estimatedDelivery.replace(/&.+?;/, ' ')] : [] // ['foo&nbsp;bar'] -> ['foo bar']

// This object should only appear if the user has selected a store.
const storeDetailsFragment = ({
  storeAddress1 = '',
  storeAddress2 = '',
  storeCity = '',
  storeState = '',
  shippingCountry = '',
  storePostCode = '',
  storeName = '',
}) => {
  if (
    !storeAddress1 &&
    !storeAddress2 &&
    !storeCity &&
    !storeState &&
    !shippingCountry &&
    !storePostCode &&
    !storeName
  ) {
    return false
  }
  return {
    address1: storeAddress1,
    address2: storeAddress2,
    city: storeCity,
    state: storeState,
    country: shippingCountry,
    postcode: storePostCode,
    storeName,
  }
}

const giftCardsFragment = (giftCardsWcs) => {
  if (!giftCardsWcs || !Array.isArray(giftCardsWcs)) return []

  return giftCardsWcs.map((giftCard) => {
    const {
      giftCardId,
      giftCardNumber,
      balance,
      amountUsed,
      remainingBalance,
    } = giftCard
    return {
      giftCardId,
      giftCardNumber,
      balance,
      amountUsed,
      remainingBalance,
    }
  })
}

const orderSummaryTransform = (
  {
    MiniBagForm,
    Basket,
    deliveryLocationTypeForm = {},
    deliveryoptionsform = false,
    OrderCalculateForm = {},
    OrderDeliveryOption = {},
    storedeliveryaddressform = {},
    deliveryaddressform = {},
    GiftCardsManagerForm = {},
    checkoutDiscountIntroEspot,
    isGuestOrder,
  } = {},
  isGuest = false,
  currencySymbol = ''
) => {
  const basket = MiniBagForm ? MiniBagForm.Basket : Basket

  const basketError = checkIfBasketObjectWithError(basket)
  if (basketError) return basketTransform(basketError, currencySymbol)

  const orderSummary = {
    isGuestOrder,
    basket: basketTransform(basket, currencySymbol),
    deliveryLocations: deliveryLocationsFragment(
      deliveryLocationTypeForm,
      deliveryoptionsform
    ),
    giftCards: giftCardsFragment(GiftCardsManagerForm.giftCards),
    deliveryInstructions: '',
    smsMobileNumber: '',
    // The following property (deliveryLocationTypeForm > deliveryCountry) has been added to handle correctly the selection of another country (e.g.: Sri Lanka).
    // The previous position (deliveryaddressform > shippingCountry) where we were grabbing the value was unchanged even in case of selection of another country.
    shippingCountry:
      path(['deliveryCountry'], deliveryLocationTypeForm) ||
      path(['shippingCountry'], deliveryaddressform) ||
      path(['deliveryDetails', 'address', 'country'], deliveryoptionsform) ||
      '', // non UK sites
    savedAddresses: Array.isArray(deliveryoptionsform.savedAddresses)
      ? deliveryoptionsform.savedAddresses.map(savedAddressFragment)
      : [],
    ageVerificationDeliveryConfirmationRequired: false,
    estimatedDelivery: estimatedDeliveryFragment(basket),
  }

  if (checkoutDiscountIntroEspot && !isEmpty(checkoutDiscountIntroEspot)) {
    orderSummary[
      espotsDesktopConstants.orderSummary.discountIntro
    ] = checkoutDiscountIntroEspot
  }

  const cardNumberHash = path(['CreditCard', 'cardNumber'], OrderCalculateForm)
  if (cardNumberHash) {
    orderSummary.cardNumberHash = cardNumberHash
  }

  const creditCard = path(['CreditCard'], OrderCalculateForm)
  if (creditCard) {
    orderSummary.creditCard = creditCardFragment(creditCard)
  }

  if (
    deliveryLocationTypeForm &&
    ['STORE', 'PARCELSHOP'].includes(
      getDeliveryType(deliveryLocationTypeForm.deliveryLocations)
    )
  ) {
    // If a registered user has not made a purchase before, these values from WCS are blank and they are not returned at all by scrAPI.
    const storeDetails = storeDetailsFragment(OrderDeliveryOption)
    if (storeDetails) {
      orderSummary.storeDetails = storeDetails
    }
    if (storedeliveryaddressform.field1) {
      orderSummary.deliveryStoreCode = storedeliveryaddressform.field1
    }
  }

  if (!isGuest) {
    const billingDetails = path(['billingDetails'], OrderCalculateForm)
    // 1st time checkout: WCS does not provide OrderCalculateForm.billingDetails and in that case the current contract
    // does not provide "billingDetails" property.
    if (billingDetails) {
      orderSummary.billingDetails = detailsFragment(billingDetails)
    }

    const deliveryDetails = deliveryDetailsFragment(deliveryoptionsform)
    if (deliveryDetails) {
      orderSummary.deliveryDetails = deliveryDetails
    }
  }

  if (isGuestOrder) {
    const guestUserEmailBillingDetails = path(
      ['billingDetails', 'email'],
      OrderCalculateForm
    )

    if (guestUserEmailBillingDetails) {
      orderSummary.email = guestUserEmailBillingDetails
    }
  }

  return orderSummary
}

export {
  deliveryPriceFragment,
  getDeliveryType,
  expressDeliveryOptionFragment,
  expressDeliveryFragment,
  deliveryMethodFragment,
  deliveryMethodsFragment,
  deliveryLocationLabelFragment,
  deliveryLocationMethodsFragment,
  deliveryLocationFragment,
  deliveryLocationsFragment,
  deliveryDetailsFragment,
  creditCardFragment,
  addressNameFragment,
  savedAddressFragment,
  estimatedDeliveryFragment,
  storeDetailsFragment,
  checkIfBasketObjectWithError,
  checkForInvalidDeliveryMethodData,
}

export default orderSummaryTransform
