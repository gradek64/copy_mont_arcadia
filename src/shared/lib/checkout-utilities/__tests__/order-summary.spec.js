import {
  removeCFSIFromOrderSummary,
  normaliseEstimatedDeliveryDate,
  normaliseDeliveryType,
  isErroredStore,
} from '../order-summary'

import { orderSummaryCollectFromStoreImmediately } from '../../../../../test/mocks/orderSummary/collect-from-store-immediately'

describe('Order Summary fix utils', () => {
  describe('removeCFSIFromOrderSummary(orderSummary)', () => {
    it('should remove Collect from Store Delivery Type from orderSummary', () => {
      expect(
        removeCFSIFromOrderSummary(orderSummaryCollectFromStoreImmediately)
      ).toMatchSnapshot()
    })
  })

  describe(normaliseEstimatedDeliveryDate.name, () => {
    it('normalises english estimated delivery date', () => {
      expect(
        normaliseEstimatedDeliveryDate(
          'no later than Thursday 14 September 2000'
        )
      ).toEqual('Thursday 14 September 2000')
    })
    it('normalises french estimated delivery date', () => {
      expect(
        normaliseEstimatedDeliveryDate(
          'pas plus tard que vendredi 15 septembre 2000'
        )
      ).toEqual('vendredi 15 septembre 2000')
    })
    it('normalises german estimated delivery date', () => {
      expect(
        normaliseEstimatedDeliveryDate(
          'nicht später als Freitag 15 September 2000'
        )
      ).toEqual('Freitag 15 September 2000')
    })
  })

  describe(`${normaliseEstimatedDeliveryDate.name} array`, () => {
    it('normalises english estimated delivery date', () => {
      expect(
        normaliseEstimatedDeliveryDate([
          'no later than Thursday 14 September 2000',
        ])
      ).toEqual('Thursday 14 September 2000')
    })
    it('normalises french estimated delivery date', () => {
      expect(
        normaliseEstimatedDeliveryDate([
          'pas plus tard que vendredi 15 septembre 2000',
        ])
      ).toEqual('vendredi 15 septembre 2000')
    })
    it('normalises german estimated delivery date', () => {
      expect(
        normaliseEstimatedDeliveryDate([
          'nicht später als Freitag 15 September 2000',
        ])
      ).toEqual('Freitag 15 September 2000')
    })
  })

  describe('normaliseDeliveryType', () => {
    it('normalises english', () => {
      expect(normaliseDeliveryType('Home Delivery Network')).toEqual(
        'Home Delivery Network'
      )
      expect(normaliseDeliveryType('Parcelnet')).toEqual('Parcelnet')
    })

    it('normalises french', () => {
      expect(normaliseDeliveryType('Express International')).toEqual(
        'Home Delivery Network'
      )
      expect(normaliseDeliveryType('Standard International')).toEqual(
        'Parcelnet'
      )
      expect(normaliseDeliveryType('Livraison Express')).toEqual(
        'Home Delivery Network'
      )
      expect(normaliseDeliveryType('Livraison Standard')).toEqual('Parcelnet')
    })

    it('normalises german', () => {
      expect(normaliseDeliveryType('Expresslieferung')).toEqual(
        'Home Delivery Network'
      )
      expect(normaliseDeliveryType('Standardlieferung')).toEqual('Parcelnet')
      expect(normaliseDeliveryType('Expresslieferung (2)')).toEqual(
        'Home Delivery Network'
      )
      expect(normaliseDeliveryType('Standardlieferung (3)')).toEqual(
        'Parcelnet'
      )
    })
  })

  describe('isErroredStore', () => {
    it('it finds no message', () => {
      expect(isErroredStore()).toEqual(false)
    })

    it('it finds different message', () => {
      expect(isErroredStore('hello')).toEqual(false)
    })

    it('it finds _ERR_DELIVERY_STORE_INVALID', () => {
      expect(isErroredStore('_ERR_DELIVERY_STORE_INVALID')).toEqual(true)
    })

    it('it finds _ERR_DELIVERY_STORE_ADDRESS_INVALID', () => {
      expect(isErroredStore('_ERR_DELIVERY_STORE_ADDRESS_INVALID')).toEqual(
        true
      )
    })

    it('it finds _ERR_DELIVERY_STORE_ADDRESS_INACTIVE', () => {
      expect(isErroredStore('_ERR_DELIVERY_STORE_ADDRESS_INACTIVE')).toEqual(
        true
      )
    })
  })
})
