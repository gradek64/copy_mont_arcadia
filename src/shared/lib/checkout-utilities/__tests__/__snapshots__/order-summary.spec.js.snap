// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order Summary fix utils removeCFSIFromOrderSummary(orderSummary) should remove Collect from Store Delivery Type from orderSummary 1`] = `
Object {
  "ageVerificationDeliveryConfirmationRequired": false,
  "basket": Object {
    "ageVerificationRequired": false,
    "deliveryOptions": Array [
      Object {
        "deliveryOptionId": 28007,
        "label": "Express / Nominated Day Delivery £6.00",
        "selected": false,
      },
      Object {
        "deliveryOptionId": 47524,
        "label": "Collect from ParcelShop £4.00",
        "selected": false,
      },
      Object {
        "deliveryOptionId": 45020,
        "label": "Collect From Store Express £3.00",
        "selected": false,
      },
      Object {
        "deliveryOptionId": 26504,
        "label": "UK Standard up to 4 working days £4.00",
        "selected": false,
      },
      Object {
        "deliveryOptionId": 45019,
        "label": "Free Collect From Store Standard £0.00",
        "selected": true,
      },
    ],
    "discounts": Array [],
    "orderId": 345682957,
    "products": Array [
      Object {
        "ageVerificationRequired": false,
        "assets": Array [
          Object {
            "assetType": "IMAGE_SMALL",
            "index": 1,
            "url": "http://media.topshop.com/wcsstore/TopShop/images/catalog/TS32G33MPNK_Thumb_F_1.jpg",
          },
          Object {
            "assetType": "IMAGE_THUMB",
            "index": 1,
            "url": "http://media.topshop.com/wcsstore/TopShop/images/catalog/TS32G33MPNK_Small_F_1.jpg",
          },
          Object {
            "assetType": "IMAGE_NORMAL",
            "index": 1,
            "url": "http://media.topshop.com/wcsstore/TopShop/images/catalog/TS32G33MPNK_2col_F_1.jpg",
          },
          Object {
            "assetType": "IMAGE_LARGE",
            "index": 1,
            "url": "http://media.topshop.com/wcsstore/TopShop/images/catalog/TS32G33MPNK_Zoom_F_1.jpg",
          },
        ],
        "attributes": Object {},
        "bundleProducts": Array [],
        "bundleSlots": Array [],
        "catEntryId": 29127435,
        "colourSwatches": Array [],
        "inStock": true,
        "isBundleOrOutfit": false,
        "items": Array [],
        "lineNumber": "32G33MPNK",
        "lowStock": true,
        "name": "GIGGLE GHILLIE Pointed Tie-Up Heel Court Shoes",
        "orderItemId": 911612200,
        "productId": 29127396,
        "quantity": 1,
        "shipModeId": 45020,
        "size": "35",
        "totalPrice": "49.00",
        "tpmLinks": Array [],
        "unitPrice": "49.00",
        "wasPrice": "",
      },
    ],
    "promotions": Array [],
    "restrictedDeliveryItem": false,
    "savedProducts": Array [],
    "subTotal": "49.00",
    "total": "52.00",
    "totalBeforeDiscount": "49.00",
  },
  "deliveryInstructions": "",
  "deliveryLocations": Array [
    Object {
      "deliveryLocationType": "HOME",
      "deliveryMethods": Array [],
      "label": "Home Delivery Standard (UK up to 4 working days; worldwide varies)  Express (UK next or nominated day; worldwide varies)",
      "selected": false,
    },
    Object {
      "deliveryLocationType": "STORE",
      "deliveryMethods": Array [
        Object {
          "additionalDescription": "Collection date Wednesday 6 September 2017",
          "deliveryOptions": Array [],
          "deliveryType": "STORE_STANDARD",
          "label": "Collect From Store Standard",
          "selected": true,
          "shipCode": "Retail Store Standard",
          "shipModeId": 45019,
        },
      ],
      "label": "Collect from Store Standard (3-7 working days) Express (next day)",
      "selected": true,
    },
  ],
  "estimatedDelivery": Array [
    "No later than Saturday 5 August 2017",
  ],
  "giftCards": Array [],
  "savedAddresses": Array [],
  "shippingCountry": "United Kingdom",
  "smsMobileNumber": "",
  "version": "1.6",
}
`;
